inference-engine/src/mkldnn_plugin/nodes/embedding_bag_sum.cpp

```diff
@@ -0,0 +1,369 @@
+        for (int i = IN_DATA_DIMS_SIZE - 1; i > 0; i--) {
+            _multipliers[i - 1] = inDataDims[i] * _multipliers[i];
+        }
+
+        if (inDataDims.size() == 2) {
+            auto jcp = jit_emb_bag_config_params();
+            jcp.emb_dim = inDataDims[1];
+            jcp.with_weights = _with_weights;
+
+            if (mayiuse(cpu::avx512_common)) {
+//                emb_bag_kernel.reset(new jit_uni_embedding_bag_sum_kernel_f32<cpu::avx512_common>(jcp));
+            } else if (mayiuse(cpu::avx2)) {
+                emb_bag_kernel.reset(new jit_uni_embedding_bag_sum_kernel_f32<cpu::avx2>(jcp));
+            } else if (mayiuse(cpu::sse42)) {
+                emb_bag_kernel.reset(new jit_uni_embedding_bag_sum_kernel_f32<cpu::sse42>(jcp));
+            }
+        }
+    } catch (InferenceEngine::details::InferenceEngineException &ex) {
+        errorMsg = ex.what();
+    }
+}
+
+StatusCode MKLDNNEmbeddingBagSum::execute(
+        std::vector<Blob::Ptr>& inputs,
+        std::vector<Blob::Ptr>& outputs,
+        ResponseDesc *resp) noexcept {
+    switch (inputs[0]->getTensorDesc().getPrecision()) {
+        case Precision::FP32: {
+            process_data<PrecisionTrait<Precision::FP32>::value_type>(inputs, outputs);
+            break;
+        }
+        case Precision::I8: {
+            process_data<PrecisionTrait<Precision::I8>::value_type>(inputs, outputs);
+            break;
+        }
+        case Precision::U8: {
+            process_data<PrecisionTrait<Precision::U8>::value_type>(inputs, outputs);
+            break;
+        }
+        case Precision::I32: {
+            process_data<PrecisionTrait<Precision::I32>::value_type>(inputs, outputs);
+            break;
+        }
+        default: {
+            if (resp) {
+                std::string errorMsg = "EmbeddingBagSum layer does not support precision '"
+                        + std::string(inputs[0]->getTensorDesc().getPrecision().name()) + "'";
+                errorMsg.copy(resp->msg, sizeof(resp->msg) - 1);
+            }
+            return GENERAL_ERROR;
+        }
+    }
+
+    return OK;
+}
+
+template<typename T>
+void MKLDNNEmbeddingBagSum::process_data(
+        std::vector<Blob::Ptr>& inputs,
+        std::vector<Blob::Ptr>& outputs) noexcept {
+    const T* src_data = inputs[0]->cbuffer().as<const T*>() +
+        inputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+    T* dst_data = outputs[0]->buffer().as<T*>() +
+        outputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+    const T* weights_data = nullptr;
+    if (_with_weights)
+        weights_data = inputs[PER_SAMPLE_WEIGHTS_IDX]->cbuffer().as<const T*>();
+    init_from_inputs(inputs);
+
+    const auto& inDataDims = inputs[0]->getTensorDesc().getDims();
+    const size_t IN_DATA_DIMS_SIZE = inDataDims.size();
+
+    const size_t OUTPUT_BAGS_NUM = outputs[0]->getTensorDesc().getDims()[0];
+
+    memset(dst_data, 0, outputs[0]->byteSize());
```

Comment:
Can we do initialization "inplace" with main "emb_cycle"? Otherwise memset assumes an extra memory pass (through all dst tensor).

Reply:
Due to there is element wise sum operation below, all elements should be initialized.